src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -222,7 +222,15 @@ def get(self, request: Request, organization: Organization) -> Response:
         project = Project.objects.get(id=int(project_id))
         enabled_features = on_demand_metrics_feature_flags(organization)
         prefilling = "organizations:on-demand-metrics-prefill" in enabled_features
-        alert_specs = get_default_version_alert_metric_specs(project, enabled_features, prefilling)
+        prefilling_for_deprecation = (
+            "organizations:on-demand-gen-metrics-deprecation-prefill" in enabled_features
+        )
+        alert_specs = get_default_version_alert_metric_specs(
+            project,
+            enabled_features,
+            prefilling,
+            prefilling_for_deprecation=prefilling_for_deprecation,
+        )
```

Comment:
in the near-ish future we're moving alert creation to a generic api, would we need to replicate this in the result or is the plan to keep this api separate?

if you want to hook into the generic api, then we could probably move this code into a serializer so we can apply it to metric based alerts.